service/history/handler.go

```diff
@@ -1144,12 +1144,36 @@ func (h *handlerImpl) SignalWithStartWorkflowExecution(
 		return nil, h.error(err1, scope, domainID, workflowID)
 	}
 
-	resp, err2 := engine.SignalWithStartWorkflowExecution(ctx, wrappedRequest)
-	if err2 != nil {
-		return nil, h.error(err2, scope, domainID, workflowID)
-	}
+	for {
```

Comment:
this looks like it could retry more than twice, would it be possible to add a counter and just check we don't end up retrying more than N times?

Reply:
Yeah, it could at most do twice. But it doesn't hurt to retry until context timeout.